mkosi/__init__.py

```diff
@@ -2751,6 +2751,45 @@ def jj(line: str) -> str:
             patch_file(os.path.join(root, 'etc/shadow'), jj)
 
 
+def set_autologin(args: CommandLineArguments, root: str, do_run_build_script: bool, for_cache: bool) -> None:
+    if do_run_build_script or for_cache or not args.autologin:
+        return
+
+    override_dir = os.path.join(root, "etc/systemd/system/console-getty.service.d")
+    os.makedirs(override_dir, mode=0o755, exist_ok=True)
+
+    override_file = os.path.join(override_dir, "autologin.conf")
+    with open(override_file, 'w') as f:
+        f.write(
+            dedent(
+                f"""
```

Comment:
I don't think this needs to be an f-string, but it could be made a raw string for less escaping further down.